src/components/users/TickForm.tsx

```diff
@@ -19,35 +21,76 @@ const TickSchema = Yup.object().shape({
     .required('Something went wrong fetching the climb Id, please try again'),
   userId: Yup.string()
     .required('Something went wrong fetching your user Id, please try again'),
-  style: Yup.string()
-    .required('Please choose an ascent style'),
-  attemptType: Yup.string()
-    .required('Please choose an ascent type'),
+  style: Yup.string(),
+  attemptType: Yup.string(),
   dateClimbed: Yup.date()
     .required('Please include a date')
     .max(new Date(), 'Please include a date in the past'),
-  grade: Yup.string()
-    .required('Something went wrong fetching the climbs grade, please try again')
+  grade: Yup.string(),
 })
+
 /**
  * Options for the dropdown comboboxes
- * Feel free to add any you think are valuable for our users
- */
-const styles = [
-  { id: 1, name: 'Solo' },
-  { id: 2, name: 'TR' },
-  { id: 3, name: 'Follow' },
-  { id: 4, name: 'Lead' },
-  { id: 5, name: 'Boulder' }
+ * Tick validation logic is complicated. see [tick_logic.md](https://github.com/OpenBeta/openbeta-graphql/blob/develop/documentation/tick_logic.md).
+ **/
+const allStyles = [
+  { id: 1, name: 'Lead' },
+  { id: 2, name: 'Follow' },
+  { id: 3, name: 'TR' },
+  { id: 4, name: 'Solo' },
+  { id: 5, name: 'Boulder' },
+  { id: 6, name: 'Aid' }
 ]
 
-const attemptTypes = [
+const allAttemptTypes = [
   { id: 1, name: 'Onsight' },
   { id: 2, name: 'Flash' },
   { id: 3, name: 'Redpoint' },
-  { id: 4, name: 'Pinkpoint' }
+  { id: 4, name: 'Pinkpoint' },
+  { id: 5, name: 'Send' },
+  { id: 6, name: 'Attempt' },
+  { id: 7, name: 'Frenchfree' }
 ]
 
+function hasKey (climbType: object, myList: string[]): boolean { return Object.keys(climbType).some(key => myList.includes(key)) }
+
+function stylesForClimbType (climbType: object): Array<{ id: number, name: string }> {
+  const leadable = hasKey(climbType, ['trad', 'sport', 'snow', 'ice', 'mixed', 'alpine'])
+  const topropeable = hasKey(climbType, ['tr']) || (leadable)
+  const aidable = hasKey(climbType, ['aid'])
+  const boulderable = hasKey(climbType, ['bouldering'])
+  const soloable = hasKey(climbType, ['deepwatersolo']) || leadable || aidable || (topropeable && !boulderable)
+
+  let styles: Array<{ id: number, name: string }> = []
+
+  if (leadable) { styles.push(...allStyles.filter(style => ['Lead', 'Follow'].includes(style.name))) }
+  if (boulderable) { styles.push(...allStyles.filter(style => ['Boulder'].includes(style.name))) }
+  if (topropeable) { styles.push(...allStyles.filter(style => ['TR'].includes(style.name))) }
+  if (aidable) { styles.push(...allStyles.filter(style => ['Aid'].includes(style.name))) }
+  if (soloable) { styles.push(...allStyles.filter(style => ['Solo'].includes(style.name))) }
+  if (styles.length === 0) { styles = [...allStyles] } // If a climb doesn't have a type, anything goes
+  styles.push({ id: 10, name: '\u00A0' })
```

Comment:
FYI the '\u00A0' is the html encoded space character. this is to get the dropdown to show a blank option (and look nice). Down below, It handles this case and sends `undefined` to the backend, which gets handles correctly in this pr https://github.com/OpenBeta/openbeta-graphql/pull/444